fendermint/testing/materializer/src/lib.rs

```diff
@@ -1,6 +1,257 @@
 // Copyright 2022-2024 Protocol Labs
 // SPDX-License-Identifier: Apache-2.0, MIT
+use multihash::MultihashDigest;
+use serde::{Deserialize, Serialize};
+use std::{
+    fmt::Display,
+    path::{Path, PathBuf},
+};
+
 pub mod manifest;
+pub mod materializer;
+pub mod testnet;
 
 #[cfg(feature = "arb")]
 mod arb;
+
+/// An ID identifying a resource within its parent.
+#[derive(Clone, Debug, Serialize, PartialEq, Eq, PartialOrd, Ord)]
+pub struct ResourceId(String);
+
+impl<'de> Deserialize<'de> for ResourceId {
+    fn deserialize<D>(deserializer: D) -> Result<Self, D::Error>
+    where
+        D: serde::Deserializer<'de>,
+    {
+        String::deserialize(deserializer).map(Self::from)
+    }
+}
```

Comment:
That's a great question. I'll should have added a comment because I forgot why I implemented it in the first place. 

I remember now: it's because the way it produces itself from a `String` is by replacing any `/` with `_` so I don't have to reject any name with errors, just silently turn it into one which is compatible with URLs.